service/create/keypair.go

```diff
@@ -0,0 +1,44 @@
+package create
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+type keyPairInput struct {
+	ec2Client   *ec2.EC2
+	clusterName string
+}
+
+func (s *Service) publicKeyMaterial() ([]byte, error) {
+	return ioutil.ReadFile(s.pubKeyFile)
+}
+
+func (s *Service) keyPair(input keyPairInput) (string, error) {
+	pkm, err := s.publicKeyMaterial()
+	if err != nil {
+		return "", err
+	}
+
+	keyPair, err := input.ec2Client.ImportKeyPair(&ec2.ImportKeyPairInput{
+		KeyName:           aws.String(input.clusterName),
+		PublicKeyMaterial: pkm,
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == "InvalidKeyPair.Duplicate" {
+			s.logger.Log("info", "keypair '%s' exists, reusing")
```

Comment:
`%s` has to be applied to something. You probably want to use the key name here?